docs/details/advisors/checks/Develop Security Checks

```diff
@@ -0,0 +1,260 @@
+                        "severity": "warning",
+                        "labels": {
+                            "current": format_version_num(num),
+                            "latest":  format_version_num(latest),
+                        },
+                    })
+ 
+                return results
+```
+
+## Check fields
+
+- **Version** (integer, required) defines what other check properties are expected, what types are supported, what is expected from the script and what it can expect from the execution environment, etc.
+- **Name** (string, required) defined machine-readable name (ID).
+- **Summary** (string, required) defines short human-readable summary.
+- **Description** (string, required) defines long human-readable description.
+- **Type** (string/enum, required) defines PMM Service type for which check is performed and query type.
+- **Query** (string, optional) contains a SQL query or MongoDB query document (as a string with proper quoting) which is executed on the PMM Client side. It may be absent if type defines the whole query by itself.
+- **Script** (string, required) contains a small Python program that processes query results, and returns check results. It is executed on the PMM Server side.
+
+Check script assumes that there is a function with a fixed name *check* that accepts a *list* of *dicts* containing returned rows for SQL databases and documents for MongoDB. It returns zero, one, or several check results that are then converted to alerts.
+
+The single query means that currently you cannot implement some advanced checks that would require several queries (and can't be implemented using SQL UNION). 
+
+Checks format and the current STT UI use different terminology for severities. Here is how different formats show up on the UI:
+
+| Format          | UI                                                                                |  |
+|-------------------|------------------------------------------------------------------------------------------|:-------------------------------------:|
+|emergency                                                    
+|alert||
+|critical|
+|error| Critical
+|warning| Major|
+|notice|Trivial|
+|info	|
+| debug	|
+
+## Backend
+
+![!](../../../_images/BackendSTT.png) 
+
+
+1. pmm-managed checks that this installation is opted-in into STT.
+2. pmm-managed downloads checks file from SaaS (see Check Service).
+3. pmm-managed verifies file signatures using a list of hard-coded public keys. At least one signature should be correct.
+4. pmm-managed sends queries to pmm-agent and gathers results.
+5. pmm-managed executes check scripts that produce alert information.
+6. pmm-managed sends alerts to Alertmanager.
+    - Due to Alertmanager design, pmm-managed has to send and re-send alerts to it much more often than the frequency of check execution. That's expected, and not important for using STT, but important for understanding how it works.
+    - Currently, Prometheus is not involved.
```

Comment:
This is probably internal information that should not affect checks develloper.

Reply:
Well I think it is relevant to who wants to develop a check to know :
- can only perform a single call (atm)
- the process internally used to execute a check 
Said that I think it is a bit confusing to mention Alert manager given we are actually not raising any alert with the advisors, but notifications.